src/libs/actions/Search.ts

```diff
@@ -57,8 +57,62 @@ function createTransactionThread(hash: number, transactionID: string, reportID:
             },
         },
     };
-
     Onyx.merge(`${ONYXKEYS.COLLECTION.SNAPSHOT}${hash}`, onyxUpdate);
 }
 
-export {search, createTransactionThread};
+function holdMoneyRequestOnSearch(searchHash: number, transactionIDList: string[], comment?: string) {
+    const optimisticData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.SNAPSHOT}${searchHash}`,
+            value: {isLoading: true},
+        },
+    ];
+    const finallyData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.SNAPSHOT}${searchHash}`,
+            value: {isLoading: false},
+        },
+    ];
```

Comment:
I agree with you, but it seems that both not having any spinner and always having a big spinner are not the best solutions.

As far as having a spinner per row this is something worth considering, but probably for another PR.
Also it's really hard to fully grasp how bad this view will look with/without spinner, since the commands are not yet implemented on backend.

My suggestion would be to move forward with this PR for now, and do any loading state changes in a separate task. Perhaps we want a more generic solution for lists.

There is also a performance concern, since doing spinner per item would mean that we need to add yet another property to every item, something like `isLoading`. I will write more about this concerns in the other comment.
For now I will leave the code as is, ping me if that's ok with you